dev/tests/integration/framework/Magento/TestFramework/Helper/Amqp.php

```diff
@@ -13,26 +13,57 @@
  */
 class Amqp
 {
+    const CONFIG_PATH_QUEUE_AMQP_HOST = 'queue/amqp/host';
+    const CONFIG_PATH_QUEUE_AMQP_USER = 'queue/amqp/user';
+    const CONFIG_PATH_QUEUE_AMQP_PASSWORD = 'queue/amqp/password';
+
     /**
      * @var Curl
      */
     private $curl;
 
+    /**
+     * @var \Magento\Framework\App\DeploymentConfig
+     */
+    private $deploymentConfig;
+
     /**
      * RabbitMQ API host
      *
      * @var string
      */
-    private $host = 'http://localhost:15672/api/';
+    private $host;
 
     /**
      * Initialize dependencies.
+     * @param \Magento\Framework\App\DeploymentConfig $deploymentConfig
      */
-    public function __construct()
-    {
+    public function __construct(
+        \Magento\Framework\App\DeploymentConfig $deploymentConfig
+    ) {
+        $this->deploymentConfig = $deploymentConfig;
         $this->curl = new Curl();
-        $this->curl->setCredentials('guest', 'guest');
+        $this->curl->setCredentials(
+            $this->deploymentConfig->get(self::CONFIG_PATH_QUEUE_AMQP_USER),
+            $this->deploymentConfig->get(self::CONFIG_PATH_QUEUE_AMQP_PASSWORD)
+        );
         $this->curl->addHeader('content-type', 'application/json');
+        $this->host = sprintf('http://%s:15672/api/', $this->deploymentConfig->get(self::CONFIG_PATH_QUEUE_AMQP_HOST));
```

Comment:
Please add similar configuration for port.

Reply:
This config doesn't currently exist. Any objection adding it so `setup:install` with the other AMQP options https://github.com/magento/magento2/blob/90c0b9867198400d2f7619d741f2eab79e0d06b9/app/code/Magento/Amqp/Setup/ConfigOptionsList.php#L69-L122 ?

Reply:
Saw it here: https://github.com/magento/magento2/blob/2.3-develop/dev/tests/integration/etc/install-config-mysql.php.dist#L20

Reply:
That is the port for the queue processor itself. The management plugin's API is a separate port configuration, but the default is `15672`

Reply:
@pmclain would be nice. Hardcoded version is good as well after you explanation 👍

Reply:
Sounds good. The config has been added to this PR. More flexibility for custom CI/CD pipelines is always a good thing.